libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts

```diff
@@ -474,19 +476,25 @@ export class SkyTimepickerComponent implements OnInit, OnDestroy {
     }
   }
 
-  #addKeydownListener(): void {
-    this.#overlayKeydownListener = fromEvent<KeyboardEvent>(
-      window.document,
-      'keydown',
-    )
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe((event) => {
-        const key = event.key?.toLowerCase();
-        /* istanbul ignore else */
-        if (key === 'escape' && this.isOpen) {
-          this.#closePicker();
-        }
-      });
+  #addKeyupListener(): void {
+    const timepickerMenuElement = this.timepickerRef?.nativeElement;
+
+    if (timepickerMenuElement) {
+      this.#overlayKeydownListener = fromEvent<KeyboardEvent>(
```

Comment:
Rename `this.#overlayKeydownListener` to `this.#overlayKeyupListener`.

Reply:
renamed